efopen/ef_generate.py

```diff
@@ -551,7 +553,7 @@ def main():
   print("aws account number: {}".format(CONTEXT.account_id))
 
   # Step through all services in the service registry
-  for CONTEXT.service in CONTEXT.service_registry.iter_services():
+  for CONTEXT.service in CONTEXT.service_registry.iter_services(CONTEXT.service):
```

Comment:
This does not work. When running the arguments specified in the PR description ef-generate fails with `RuntimeError: service registry: ellation_formation/service_registry.json doesn't have 'eec' section listed in EFConfig`.

`iter_services` looks for service groups, not specific services. Service groups are the top level keys in `service_registry.json`.

Not only that, this code is very bad practice. It gives the same variable dual meaning and is confusing.

Reply:
In order for this to work, you would have to do something like
```suggestion
  for CONTEXT.service in CONTEXT.service_registry.iter_services():
     if not service_name.startswith(selected_service):
        continue
```
Please do not reuse `CONTEXT.service` for the selected service. The code has enough dual meanings as is.